README

Kano-Credits

Uses python curses.

To build program so the text and art are in line, run:

python build.py

Then to run the program:

python main.py

To add ascii art, save in a text file - for credits for staff, preferably put in 18-20 line intervals including the ascii art and one line of white space directly below.  Will need to repeat the art at least once (see examples in staff-images folder).

Create an ascii object in ascii.py, and list the art in an array e.g. in staffAsciiArt.

Similarly add the file containing the text (list of names) in the text.py file.
Currently the list of names is in the format:

Name1;Information about person1
Name2;Information about person2

The build progarm creates a file to format the text like:

Name
Information about person1






Name2
Information about person2

with an appropriate amount of whitespace to match the art and text together.

Currently formatted for a 1024 by 768 window.

debian/changelog
kano-credits (1.1-1) unstable; urgency=low

  * Added backers and blur

 -- Team Kano <dev@kano.me>  Mon, 16 Jun 2014 11:02:31 +0000

kano-credits (1.0-27) unstable; urgency=low

  * Done some refactoring

 -- Team Kano <dev@kano.me>  Wed, 14 May 2014 11:02:31 +0000

kano-credits (1.0-26) unstable; urgency=low

  * Package rebuilt, updated to revision 1d911cd9.

 -- Team Kano <dev@kano.me>  Fri, 14 Feb 2014 15:02:31 +0000

kano-credits (1.0-25) unstable; urgency=low

  * Package rebuilt, updated to revision 6ca2f269.

 -- Team Kano <dev@kano.me>  Thu, 06 Feb 2014 02:16:55 +0000

kano-credits (1.0-24) unstable; urgency=low

  * Package rebuilt, updated to revision a482536d.

 -- Team Kano <dev@kano.me>  Wed, 05 Feb 2014 02:15:07 +0000

kano-credits (1.0-23) unstable; urgency=low

  * Package rebuilt, updated to revision b2c42214.

 -- Team Kano <dev@kano.me>  Sat, 11 Jan 2014 02:17:24 +0000

kano-credits (1.0-22) unstable; urgency=low

  * Package rebuilt, updated to revision 02a71618.

 -- Team Kano <dev@kano.me>  Thu, 02 Jan 2014 02:28:04 +0000

kano-credits (1.0-21) unstable; urgency=low

  * Package rebuilt, updated to revision d105b289.

 -- Team Kano <dev@kano.me>  Wed, 01 Jan 2014 02:27:54 +0000

kano-credits (1.0-20) unstable; urgency=low

  * Package rebuilt, updated to revision cb30414c.

 -- Team Kano <dev@kano.me>  Tue, 31 Dec 2013 02:28:02 +0000

kano-credits (1.0-19) unstable; urgency=low

  * Package rebuilt, updated to revision 9df03933.

 -- Team Kano <dev@kano.me>  Mon, 30 Dec 2013 02:27:24 +0000

kano-credits (1.0-18) unstable; urgency=low

  * Package rebuilt, updated to revision 7baec406.

 -- Team Kano <dev@kano.me>  Sun, 29 Dec 2013 02:27:30 +0000

kano-credits (1.0-17) unstable; urgency=low

  * Package rebuilt, updated to revision e532258a.

 -- Team Kano <dev@kano.me>  Sat, 28 Dec 2013 02:27:28 +0000

kano-credits (1.0-16) unstable; urgency=low

  * Package rebuilt, updated to revision 81cfb024.

 -- Team Kano <dev@kano.me>  Fri, 27 Dec 2013 02:27:18 +0000

kano-credits (1.0-15) unstable; urgency=low

  * Package rebuilt, updated to revision c1664091.

 -- Team Kano <dev@kano.me>  Thu, 26 Dec 2013 02:27:09 +0000

kano-credits (1.0-14) unstable; urgency=low

  * Package rebuilt, updated to revision 3d7e2084.

 -- Team Kano <dev@kano.me>  Wed, 25 Dec 2013 02:27:46 +0000

kano-credits (1.0-13) unstable; urgency=low

  * Package rebuilt, updated to revision 2eec2dce.

 -- Team Kano <dev@kano.me>  Tue, 24 Dec 2013 02:27:51 +0000

kano-credits (1.0-12) unstable; urgency=low

  * Package rebuilt, updated to revision 50e4b1dc.

 -- Team Kano <dev@kano.me>  Mon, 23 Dec 2013 02:27:29 +0000

kano-credits (1.0-11) unstable; urgency=low

  * Package rebuilt, updated to revision da65b6d6.

 -- Team Kano <dev@kano.me>  Sun, 22 Dec 2013 02:27:28 +0000

kano-credits (1.0-10) unstable; urgency=low

  * Package rebuilt, updated to revision ec23e288.

 -- Team Kano <dev@kano.me>  Sat, 21 Dec 2013 02:27:22 +0000

kano-credits (1.0-9) unstable; urgency=low

  * Package rebuilt, updated to revision 58c01e65.

 -- Team Kano <hello@kano.me>  Fri, 20 Dec 2013 13:32:43 +0000

kano-credits (1.0-8) unstable; urgency=low

  * Package rebuilt, updated to revision cd088fbb.

 -- Team Kano <hello@kano.me>  Fri, 20 Dec 2013 12:01:34 +0000

kano-credits (1.0-7) unstable; urgency=low

  * Package rebuilt, updated to revision b8ef752b.

 -- Team Kano <hello@kano.me>  Fri, 20 Dec 2013 11:49:47 +0000

kano-credits (1.0-6) unstable; urgency=low

  * Package rebuilt, updated to revision b15cdd4e.

 -- Team Kano <hello@kano.me>  Fri, 20 Dec 2013 10:51:40 +0000

kano-credits (1.0-5) unstable; urgency=low

  * Package rebuilt, updated to revision 2167ee3e.

 -- Team Kano <hello@kano.me>  Fri, 20 Dec 2013 10:12:39 +0000

kano-credits (1.0-4) unstable; urgency=low

  * Package rebuilt, updated to revision 9fb1b94f.

 -- Team Kano <hello@kano.me>  Fri, 20 Dec 2013 09:20:50 +0000

kano-credits (1.0-3) unstable; urgency=low

  * Package rebuilt, updated to revision 6de19100.

 -- Team Kano <hello@kano.me>  Fri, 20 Dec 2013 08:48:07 +0000

kano-credits (1.0-2) unstable; urgency=low

  * Package rebuilt, updated to revision 7ef0b93b.

 -- Team Kano <hello@kano.me>  Thu, 19 Dec 2013 22:56:48 +0000

kano-credits (1.0-1.1) unstable; urgency=low

  [ Alex ]
  * Initial release. (Closes: #XXXXXX)

  [ Team Kano ]
  * Package rebuilt, updated to revision 0a4ff19b.

 -- Team Kano <hello@kano.me>  Thu, 19 Dec 2013 22:05:49 +0000


ascii.py

# ascii.py
#
# Copyright (C) 2013 Kano Computing Ltd.
# Licence:  http://www.gnu.org/licences/gpl-2.0.txt GNU General Public License v2
#

import curses
import console
import graphics

visibleAsciiArt = []
staffAsciiArt = []
premiumAsciiArt = []
allAsciiArt = []
height = 0
width = 0
printMessageToScreen = ''


class AsciiArt(object):

    # art will store the asciiArt
    # name will store the name of the file
    # numberOfLines will store how long each piece of ascii art is
    # startingHeight gives us how high on the screen the image is drawn.
    # When this value is < 2 - numberOfLines, it gets removed from the visibleAsciiArt array
    
    def __init__(self, name, numberOfLines, color):
        global height
        global width

        (width, height) = console.getTerminalSize()

        self.name = name
        self.numberOfLines = numberOfLines
        self.startingHeight = height
        self.currentHeight = height
        self.color = color

        with open(self.name) as f:
            self.art = f.readlines()
        for j in range(len(self.art)):
            self.art[j] = self.art[j].rstrip('\n')

    # This function draws the ascii art on the terminal and displays it
    #
    #
    def drawArt(self):
        global width

        x = width / 20
        y = self.currentHeight
        tempy = self.currentHeight
        idx = 0
        idx2 = self.numberOfLines

        # if element is a set distance above starting point, and staffAsciiArt is not empty
        if (y == (self.startingHeight - self.numberOfLines)) and staffAsciiArt:
            makeStaffElementVisible()
        # if element from premiumAsciiArt is a set distance from where it started and is in the premiumA
        elif (y == (self.startingHeight - self.numberOfLines)) and self in premiumAsciiArt:
            switchToPremiumImages()

        # if image would otherwise be drawn outside (above) the terminal
        if y <= 0:
            tempy = 0
            idx = -y

        # if image would otherwise be drawn outside (below) the terminal
        if y >= height - self.numberOfLines - 1:
            idx2 = height - y

        for idx in range(idx, idx2):
            if y == -self.numberOfLines:
                # so the element is back at the bottom of the page at the end of this function
                self.currentHeight = self.startingHeight
                return
            graphics.drawTile(x, tempy, self.art[idx], self.color)
            tempy = tempy + 1

        self.currentHeight = self.currentHeight - 1


def init():
    global visibleAsciiArt
    global premiumAsciiArt
    global staffAsciiArt
    global allAsciiArt

    backers = AsciiArt("headings/backers", 9, curses.color_pair(8))
    specialThanks = AsciiArt("headings/specialThanks", 12, curses.color_pair(8))
    translators = AsciiArt("headings/translators", 9, curses.color_pair(8))
    kano = AsciiArt("headings/kano", 21, curses.color_pair(8))

    brainy = AsciiArt("staff-images/brainy-guy", 21, curses.color_pair(5))
    hand = AsciiArt("staff-images/hand", 15, curses.color_pair(2))
    book = AsciiArt("staff-images/book", 18, curses.color_pair(3))
    monkey = AsciiArt("staff-images/monkey", 24, curses.color_pair(4))
    hair = AsciiArt("staff-images/hair", 18, curses.color_pair(6))
    pikachu = AsciiArt("staff-images/pikachu", 22, curses.color_pair(6))
    bowl = AsciiArt("staff-images/bowl", 18, curses.color_pair(7))
    camera = AsciiArt("staff-images/camera", 18, curses.color_pair(8))
    wizard = AsciiArt("staff-images/wizard", 18, curses.color_pair(2))
    shoes = AsciiArt("staff-images/shoes", 18, curses.color_pair(8))
    coffee = AsciiArt("staff-images/coffee", 18, curses.color_pair(5))
    boom = AsciiArt("staff-images/boom", 15, curses.color_pair(4))
    smileyFace = AsciiArt("staff-images/smiley-face", 15, curses.color_pair(6))
    peter = AsciiArt("staff-images/peter-pan", 26, curses.color_pair(2))
    aeroplane = AsciiArt("staff-images/aeroplane", 18, curses.color_pair(3))

    imageBlank1 = AsciiArt("headings/blank", 16, curses.color_pair(8))
    imageBlank2 = AsciiArt("headings/blank", 16, curses.color_pair(8))
    imageBlank3 = AsciiArt("headings/blank", 16, curses.color_pair(8))
    blank = AsciiArt("staff-images/blank", 1, curses.color_pair(6))

    """allAsciiArt = [software, hand, brainy, coffee, boom, smileyFace, hair, design, book, peter, monkey, pikachu,
                    bowl, camera, wizard, shoes, aeroplane, specialThanks, imageBlank1, translators, imageBlank2,
                    backers, imageBlank3, blank]"""
    staffAsciiArt = [ brainy, hand, hair, coffee, boom, smileyFace, book, peter, monkey, pikachu,
                    bowl, camera, wizard, shoes, aeroplane, specialThanks, imageBlank3, translators, imageBlank1,
                    backers, imageBlank2, blank ]
    visibleAsciiArt = [kano]
    allAsciiArt = visibleAsciiArt + staffAsciiArt

    visibleAsciiArt[0].currentHeight = visibleAsciiArt[0].currentHeight - visibleAsciiArt[0].numberOfLines


def makeStaffElementVisible():
    global visibleAsciiArt
    global staffAsciiArt

    visibleAsciiArt.append(staffAsciiArt[0])
    staffAsciiArt.pop(0)


def switchToPremiumImages():  # TODO: need to turn this off if filled with icons already
    global visibleAsciiArt
    global premiumAsciiArt
    global printMessageToScreen

    if len(premiumAsciiArt) != 0:
        premiumAsciiArt[0].currentHeight = premiumAsciiArt[0].startingHeight
        visibleAsciiArt.append(premiumAsciiArt[0])
        premiumAsciiArt.append(premiumAsciiArt[0])
        premiumAsciiArt.pop(0)


# Sticks the object at the top of the array onto the bottom
def moveObjectToBottom():
    global visibleAsciiArt
    visibleAsciiArt[0].currentHeight = visibleAsciiArt[0].startingHeight
    visibleAsciiArt.append(visibleAsciiArt[0])
    visibleAsciiArt.remove(visibleAsciiArt[0])


def removeTopElement():
    global visibleAsciiArt
    visibleAsciiArt.pop(0)
